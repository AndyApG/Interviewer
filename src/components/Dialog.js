/* eslint-disable */
/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Button, Icon, animation } from 'network-canvas-ui';
import { CSSTransitionGroup } from 'react-transition-group';
import xss from 'xss';

/**
  * Renders a dialog box.
  */
const Dialog = (props) => {
  const {
    title,
    children,
    show,
    hasCancelButton,
    additionalInformation,
    cancelLabel,
    confirmLabel,
    onCancel,
    onConfirm,
    type,
  } = props;

  let cancelButton = null;
  if (hasCancelButton) {
    cancelButton = <Button color="navy-taupe" onClick={onCancel} content={cancelLabel} />;
  }

  const typeColor = {
    'info': 'primary',
    'warning': 'mustard',
    'error': 'neon-coral',
  }

  function createMarkup() {
    const safeString = xss(additionalInformation, {
      whiteList: {
        h3: [],
        p: [],
        ul: [],
        li: [],
      },
      stripIgnoreTag: true,
    });
    return {
      __html: safeString
    };
  };


  let dialogClasses  = cx('dialog__window dialog__window--' + type);
  let additionalTextarea = additionalInformation ? <div className="dialog__additional-box" dangerouslySetInnerHTML={createMarkup()} />: '';

  return (
    <CSSTransitionGroup
      transitionName="dialog--transition"
      transitionEnterTimeout={animation.duration.standard}
      transitionLeaveTimeout={animation.duration.standard}
    >
      { show &&
        <div key="dialog" className="dialog">
          <div className={dialogClasses} onClick={e => e.stopPropagation()}>
            <div className="dialog__main">
              <div className="dialog__main-icon">
                <Icon name={type} />
              </div>
              <div className="dialog__main-content">
                <h2 className="dialog__main-title">{title}</h2>
                {children}
              </div>
            </div>
            <div className="dialog__additional-content">
              {additionalTextarea}
            </div>
            <footer className="dialog__footer">
              { cancelButton }
              <Button onClick={onConfirm} color={typeColor[type]} content={confirmLabel} />
            </footer>
          </div>
        </div>
      }
    </CSSTransitionGroup>
  );
};

Dialog.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any,
  show: PropTypes.bool,
  type: PropTypes.string.isRequired,
  hasCancelButton: PropTypes.bool.isRequired,
  confirmLabel: PropTypes.string.isRequired,
  additionalInformation: PropTypes.string,
  cancelLabel: PropTypes.string.isRequired,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

Dialog.defaultProps = {
  children: null,
  additionalInformation: null,
  show: false,
};

export default Dialog;
