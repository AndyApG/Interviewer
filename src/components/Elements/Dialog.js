/* eslint-disable */
/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Button, Icon, animation } from 'network-canvas-ui';
import { CSSTransitionGroup } from 'react-transition-group';

/**
  * Renders a dialog box.
  */
const Dialog = (props) => {
  const {
    title,
    children,
    show,
    hasCancelButton,
    additionalInformation,
    cancelLabel,
    confirmLabel,
    onCancel,
    onConfirm,
    type,
  } = props;

  let cancelButton = null;
  if (hasCancelButton) {
    cancelButton = <Button color="navy-taupe" onClick={onCancel} content={cancelLabel} />;
  }

  const typeColor = {
    'info': 'primary',
    'warning': 'mustard',
    'error': 'neon-coral',
  }

  let dialogClasses  = cx('dialog__window dialog__window--' + type);
  let additionalTextarea = additionalInformation ? <div className="dialog__layout-additional-info" dangerouslySetInnerHTML={{ __html: additionalInformation }} />: '';

  return (
    <CSSTransitionGroup
      transitionName="dialog--transition"
      transitionEnterTimeout={animation.duration.standard}
      transitionLeaveTimeout={animation.duration.standard}
    >
      { show &&
        <div key="dialog" className="dialog">
          <div className={dialogClasses} onClick={e => e.stopPropagation()}>
            <div className="dialog__layout">
              <div className="dialog__layout-icon">
                <Icon name={type} />
              </div>
              <div className="dialog__layout-content">
                <h2 className="dialog__layout-title">{title}</h2>
                {children}
                {additionalTextarea}
              </div>
            </div>
            <footer className="dialog__footer">
              { cancelButton }
              <Button onClick={onConfirm} color={typeColor[type]} content={confirmLabel} />
            </footer>
          </div>
        </div>
      }
    </CSSTransitionGroup>
  );
};

Dialog.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any,
  show: PropTypes.bool,
  type: PropTypes.string.isRequired,
  hasCancelButton: PropTypes.bool.isRequired,
  confirmLabel: PropTypes.string.isRequired,
  additionalInformation: PropTypes.string,
  cancelLabel: PropTypes.string.isRequired,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

Dialog.defaultProps = {
  children: null,
  additionalInformation: null,
  show: false,
};

export default Dialog;
