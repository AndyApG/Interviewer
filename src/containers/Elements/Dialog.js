/* eslint-disable no-shadow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog as DialogComponent } from '../../components/Elements';
import modal from '../../behaviours/modal';

/**
  * A modal window which can be toggled open an closed.
  * @extends Component
  */
class Dialog extends Component {

  confirmModal = () => {
    this.props.close();
    this.props.onConfirm();
  };

  cancelModal = () => {
    this.props.close();
    this.props.onCancel();
  };

  render() {
    const {
      title,
      children,
      show,
      hasCancelButton,
      type,
    } = this.props;

    return (
      <DialogComponent
        show={show}
        title={title}
        hasCancelButton={hasCancelButton}
        type={type}
        onConfirm={this.confirmModal}
        onCancel={this.cancelModal}
      >
        {children}
      </DialogComponent>
    );
  }
}

Dialog.propTypes = {
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  hasCancelButton: PropTypes.bool,
  show: PropTypes.bool,
  children: PropTypes.any,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
};

Dialog.defaultProps = {
  show: false,
  hasCancelButton: true,
  children: null,
};


export default modal(Dialog);
