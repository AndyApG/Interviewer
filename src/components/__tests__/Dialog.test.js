/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../Dialog';

const testProps = {
  title: '',
  type: '',
  name: '',
  confirmLabel: '',
  onConfirm: () => {},
};

describe('<Dialog /> component', () => {
  it('can render', () => {
    const component = shallow(
      <Dialog {...testProps} />,
    );

    expect(component).toMatchSnapshot();
  });

  describe('cancel button', () => {
    it('no cancel button renders by default', () => {
      const component = shallow(
        <Dialog {...testProps} />,
      );

      expect(component.find('.dialog__footer Button').length).toEqual(1);
    });

    it('when hasCancelButton is true, cancel button renders', () => {
      const component = shallow(
        <Dialog {...{ ...testProps, hasCancelButton: true }} />,
      );

      expect(component.find('.dialog__footer Button').length).toEqual(2);
    });
  });

  describe('additional information', () => {
    it('no additional information renders by default', () => {
      const component = shallow(
        <Dialog {...testProps} />,
      );

      expect(component.find('.dialog__additional-content').length).toEqual(0);
    });

    it('when additional information is set, it renders', () => {
      const component = shallow(
        <Dialog {...{ ...testProps, additionalInformation: 'test' }} />,
      );

      expect(component.find('.dialog__additional-content').length).toEqual(1);
    });
  });
});
