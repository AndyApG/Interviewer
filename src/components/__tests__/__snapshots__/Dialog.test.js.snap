// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> component can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dialog
    additionalInformation={null}
    cancelLabel=""
    confirmLabel=""
    hasCancelButton={false}
    name=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title=""
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="dialog"
      >
        <div
          className="dialog__background"
          transition-role="background"
        />
        <div
          className="dialog__window dialog__window--"
          onClick={[Function]}
          transition-role="window"
        >
          <div
            className="dialog__main"
          >
            <div
              className="dialog__main-icon"
            >
              <Icon
                className=""
                color=""
                name=""
                style={Object {}}
              />
            </div>
            <div
              className="dialog__main-content"
            >
              <h2
                className="dialog__main-title"
              >
                
              </h2>
            </div>
          </div>
          <footer
            className="dialog__footer"
          >
            <Button
              color=""
              content=""
              icon=""
              iconPosition="right"
              onClick={[Function]}
              size=""
            />
          </footer>
        </div>
      </div>,
      "in": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "dialog",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dialog__background"
            transition-role="background"
          />,
          <div
            className="dialog__window dialog__window--"
            onClick={[Function]}
            transition-role="window"
          >
            <div
              className="dialog__main"
            >
              <div
                className="dialog__main-icon"
              >
                <Icon
                  className=""
                  color=""
                  name=""
                  style={Object {}}
                />
              </div>
              <div
                className="dialog__main-content"
              >
                <h2
                  className="dialog__main-title"
                >
                  
                </h2>
              </div>
            </div>
            <footer
              className="dialog__footer"
            >
              <Button
                color=""
                content=""
                icon=""
                iconPosition="right"
                onClick={[Function]}
                size=""
              />
            </footer>
          </div>,
        ],
        "className": "dialog",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dialog__background",
            "transition-role": "background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dialog__main"
              >
                <div
                  className="dialog__main-icon"
                >
                  <Icon
                    className=""
                    color=""
                    name=""
                    style={Object {}}
                  />
                </div>
                <div
                  className="dialog__main-content"
                >
                  <h2
                    className="dialog__main-title"
                  >
                    
                  </h2>
                </div>
              </div>,
              null,
              <footer
                className="dialog__footer"
              >
                <Button
                  color=""
                  content=""
                  icon=""
                  iconPosition="right"
                  onClick={[Function]}
                  size=""
                />
              </footer>,
            ],
            "className": "dialog__window dialog__window--",
            "onClick": [Function],
            "transition-role": "window",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="dialog__main-icon"
                  >
                    <Icon
                      className=""
                      color=""
                      name=""
                      style={Object {}}
                    />
                  </div>,
                  <div
                    className="dialog__main-content"
                  >
                    <h2
                      className="dialog__main-title"
                    >
                      
                    </h2>
                  </div>,
                ],
                "className": "dialog__main",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Icon
                      className=""
                      color=""
                      name=""
                      style={Object {}}
                    />,
                    "className": "dialog__main-icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "",
                      "color": "",
                      "name": "",
                      "style": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h2
                        className="dialog__main-title"
                      >
                        
                      </h2>,
                      null,
                    ],
                    "className": "dialog__main-content",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "dialog__main-title",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "h2",
                    },
                    null,
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  null,
                  <Button
                    color=""
                    content=""
                    icon=""
                    iconPosition="right"
                    onClick={[Function]}
                    size=""
                  />,
                ],
                "className": "dialog__footer",
              },
              "ref": null,
              "rendered": Array [
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": null,
                    "color": "",
                    "content": "",
                    "icon": "",
                    "iconPosition": "right",
                    "onClick": [Function],
                    "size": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "footer",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="dialog"
        >
          <div
            className="dialog__background"
            transition-role="background"
          />
          <div
            className="dialog__window dialog__window--"
            onClick={[Function]}
            transition-role="window"
          >
            <div
              className="dialog__main"
            >
              <div
                className="dialog__main-icon"
              >
                <Icon
                  className=""
                  color=""
                  name=""
                  style={Object {}}
                />
              </div>
              <div
                className="dialog__main-content"
              >
                <h2
                  className="dialog__main-title"
                >
                  
                </h2>
              </div>
            </div>
            <footer
              className="dialog__footer"
            >
              <Button
                color=""
                content=""
                icon=""
                iconPosition="right"
                onClick={[Function]}
                size=""
              />
            </footer>
          </div>
        </div>,
        "in": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "dialog",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dialog__background"
              transition-role="background"
            />,
            <div
              className="dialog__window dialog__window--"
              onClick={[Function]}
              transition-role="window"
            >
              <div
                className="dialog__main"
              >
                <div
                  className="dialog__main-icon"
                >
                  <Icon
                    className=""
                    color=""
                    name=""
                    style={Object {}}
                  />
                </div>
                <div
                  className="dialog__main-content"
                >
                  <h2
                    className="dialog__main-title"
                  >
                    
                  </h2>
                </div>
              </div>
              <footer
                className="dialog__footer"
              >
                <Button
                  color=""
                  content=""
                  icon=""
                  iconPosition="right"
                  onClick={[Function]}
                  size=""
                />
              </footer>
            </div>,
          ],
          "className": "dialog",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__background",
              "transition-role": "background",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="dialog__main"
                >
                  <div
                    className="dialog__main-icon"
                  >
                    <Icon
                      className=""
                      color=""
                      name=""
                      style={Object {}}
                    />
                  </div>
                  <div
                    className="dialog__main-content"
                  >
                    <h2
                      className="dialog__main-title"
                    >
                      
                    </h2>
                  </div>
                </div>,
                null,
                <footer
                  className="dialog__footer"
                >
                  <Button
                    color=""
                    content=""
                    icon=""
                    iconPosition="right"
                    onClick={[Function]}
                    size=""
                  />
                </footer>,
              ],
              "className": "dialog__window dialog__window--",
              "onClick": [Function],
              "transition-role": "window",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="dialog__main-icon"
                    >
                      <Icon
                        className=""
                        color=""
                        name=""
                        style={Object {}}
                      />
                    </div>,
                    <div
                      className="dialog__main-content"
                    >
                      <h2
                        className="dialog__main-title"
                      >
                        
                      </h2>
                    </div>,
                  ],
                  "className": "dialog__main",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Icon
                        className=""
                        color=""
                        name=""
                        style={Object {}}
                      />,
                      "className": "dialog__main-icon",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "",
                        "color": "",
                        "name": "",
                        "style": Object {},
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h2
                          className="dialog__main-title"
                        >
                          
                        </h2>,
                        null,
                      ],
                      "className": "dialog__main-content",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "",
                          "className": "dialog__main-title",
                        },
                        "ref": null,
                        "rendered": "",
                        "type": "h2",
                      },
                      null,
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    null,
                    <Button
                      color=""
                      content=""
                      icon=""
                      iconPosition="right"
                      onClick={[Function]}
                      size=""
                    />,
                  ],
                  "className": "dialog__footer",
                },
                "ref": null,
                "rendered": Array [
                  null,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": null,
                      "color": "",
                      "content": "",
                      "icon": "",
                      "iconPosition": "right",
                      "onClick": [Function],
                      "size": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "footer",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
